test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/BigDecimalGenerator.java

```
@@ -1,23 +1,56 @@
 
-import java.math.BigDecimal;
+import static java.lang.String.format;
+import static java.math.BigDecimal.ROUND_HALF_EVEN;
 
-import static java.lang.Math.pow;
-import static java.math.RoundingMode.FLOOR;
+import java.math.BigDecimal;
 
 public class BigDecimalGenerator extends Generator<BigDecimal> {
 
-    private final Integer max;
-    private final Integer decimalPlaces;
+    private static final long DEFAULT_MIN = -1000000000L;
+    private static final long DEFAULT_MAX = 1000000000L;
+    private static final int DEFAULT_SCALE = 2;
+
+    private final BigDecimal min;
+    private final BigDecimal max;
+    private final Integer scale;
+
+    public BigDecimalGenerator() {
+        min = new BigDecimal(DEFAULT_MIN);
+        max = new BigDecimal(DEFAULT_MAX);
+        scale = DEFAULT_SCALE;
+    }
+
+    public BigDecimalGenerator(final Integer min, final Integer max, final Integer scale) {
+        this.min = new BigDecimal(min);
+        this.max = new BigDecimal(max);
+        this.scale = scale;
+    }
 
-    public BigDecimalGenerator(final Integer max, final Integer decimalPlaces) {
+    public BigDecimalGenerator(final BigDecimal min, final BigDecimal max, final Integer scale) {
+        this.min = min;
         this.max = max;
-        this.decimalPlaces = decimalPlaces;
+        this.scale = scale;
     }
 
     @Override
     public BigDecimal next() {
-        final double pow = pow(10, decimalPlaces);
-        return new BigDecimal((long) RANDOM.nextInt((int) (max * pow)) / pow).setScale(decimalPlaces, FLOOR);
+        return generateRandomBigDecimalBetween(min, max).setScale(scale, ROUND_HALF_EVEN);
+    }
+
+    private BigDecimal generateRandomBigDecimalBetween(final BigDecimal min, final BigDecimal max) {
+        if (min.compareTo(max) >= 0) {
+            throw new IllegalArgumentException(format("Min value cannot be greater than or equal to Max value, got Min: %s and Max: %s", min, max));
+        }
+        if (scale < 0) {
```

Moved to constructor to fail fast